phoenix-android/src/main/kotlin/fr/acinq/phoenix/android/init/RestoreWalletView.kt

```diff
@@ -41,88 +49,229 @@ import fr.acinq.phoenix.controllers.init.RestoreWallet
+            subtitle = stringResource(id = R.string.restore_instructions)
+        )
 
-    val keyState = produceState<KeyState>(initialValue = KeyState.Unknown, true) {
-        value = SeedManager.getSeedState(context)
-    }
+        val log = logger("RestoreWallet")
+        val context = LocalContext.current
+
+        val vm: InitViewModel =
+            viewModel(factory = InitViewModel.Factory(controllerFactory, CF::initialization))
+
+        val keyState = produceState<KeyState>(initialValue = KeyState.Unknown, true) {
+            value = SeedManager.getSeedState(context)
+        }
+
+        var filteredWords = remember { listOf<String>() }
+        val selectedWords = vm.selectedWords
+
+        when (keyState.value) {
+            is KeyState.Absent -> {
+                MVIView(CF::restoreWallet) { model, postIntent ->
+                    Column(
+                        modifier = Modifier
+                            .padding(start = 24.dp, bottom = 24.dp, end = 24.dp)
+                            .fillMaxWidth()
+                    ) {
 
-    when (keyState.value) {
-        is KeyState.Absent -> {
-            MVIView(CF::restoreWallet) { model, postIntent ->
-                Column(
-                    modifier = Modifier
-                        .padding(24.dp)
-                        .fillMaxWidth()
-                ) {
-                    var wordsInput by remember { mutableStateOf("") }
-                    when (model) {
-                        is RestoreWallet.Model.Ready -> {
-                            var showDisclaimer by remember { mutableStateOf(true) }
-                            var hasCheckedWarning by remember { mutableStateOf(false) }
-                            if (showDisclaimer) {
-                                Text(stringResource(R.string.restore_disclaimer_message))
-                                Row(Modifier.padding(vertical = 24.dp), verticalAlignment = Alignment.CenterVertically) {
-                                    Checkbox(hasCheckedWarning, onCheckedChange = { hasCheckedWarning = it })
-                                    Spacer(Modifier.width(16.dp))
-                                    Text(stringResource(R.string.restore_disclaimer_checkbox))
+                        var wordsInput by remember { mutableStateOf("") }
+
+                        TextInput(
+                            text = wordsInput,
+                            onTextChange = {
+                                wordsInput = if (it.contains(" ")) {
+                                    if (filteredWords.count() > 0) {
+                                        vm.addWord(filteredWords[0])
+                                    }
+                                    ""
+                                } else {
+                                    it
+                                }
+                                postIntent(RestoreWallet.Intent.FilterWordList(predicate = wordsInput))
```

Comment:
If we already have 12 words in the `vm.selectedWords` list then we still add words but the import button will be disabled.

Reply:
Done in 57dac620